dbapi/src/main/java/com/griddynamics/jagger/dbapi/provider/SessionInfoProviderImpl.java

```diff
@@ -273,53 +283,61 @@ public Long getFirstPosition(Set<String> selectedIds) throws RuntimeException {
         checkArgument(length >= 0, "length is negative");
         checkNotNull(sessionIds, "sessionIds is null");
 
-        long timestamp = System.currentTimeMillis();
+        boolean isFetchAll = sessionIds.size() == 0; // if input set is empty - fetch all session data.
 
+        long timestamp = System.currentTimeMillis();
         List<SessionDataDto> sessionDataDtoList;
-
         try {
+            Query query = entityManager.createQuery(
+                    "select sd from SessionData as sd " +
+                    (isFetchAll ? " " : "where sd.sessionId in (:sessionIds) ") +
+                    "order by sd.startTime asc"
+            ).setFirstResult(start);
+            if (!isFetchAll) {
+                query.setParameter("sessionIds", new ArrayList<String>(sessionIds))
+                     .setMaxResults(length);
+            }
```

Comment:
(1).
The approach with two if clauses is fine. And you can go ahead with committing.
But you can look through Criteria API. May be the code will take less space and become more clear. 
[Using the Criteria API to Create Queries](http://docs.oracle.com/javaee/6/tutorial/doc/gjitv.html)